src/OpenRiaServices.Tools/Framework/DataContractProxyGenerator.cs

```diff
@@ -549,8 +536,11 @@ private IEnumerable<Attribute> FilterTypeAttributes(AttributeCollection typeAttr
                 }
             }
 
-            // Filter out attributes in filteredAttributes and attributeTypesToFilter
-            return typeAttributes.Cast<Attribute>().Where(a => !_attributeTypesToFilter.Contains(a.GetType()) && !filteredAttributes.Contains(a));
+            // Filter out attributes in filteredAttributes, attributes which should only be used by the compiler,
+            // DataContractAttribute and KnownTypeAttribute since they are handled seperatly
+            return typeAttributes.Cast<Attribute>().Where(a => a.GetType().Namespace != "System.Runtime.CompilerServices"
+            && a.GetType().Name is not ("DataContractAttribute" or "KnownTypeAttribute")
```

Comment:
fyi: `a.GetType() != typeof(DataContractAttribute) && ..:` should be slightly faster, but without measuring I would not expect it to make a large difference, maybe some ms  for total code generation (it becomes a single "pointer compare" instead of string comparison). 

So either way should be fine.

Reply:
If you decide on string comparison it might be better to compare against FUllName (including namespace) instead to avoid matching on any user defined attribute in another namespace.

Reply:
I did that comparison you suggest but that generates codesmells which blocks completion so I "fixed" it to this. Would it be better to do `a.GetType() is typeof(DataContractAttribute) && ...` or `a.GetType() is not(typeof(...) or ...` instead of this?

Reply:
I would say your original commit was "better". 
It checks exakt type (not just same name) and does it more efficiently. 

I recommend you either remove the last commit or revert it and that we complete it anyway. 

The code analysis has it flaws and can provide good suggestions, but it needs to be questioned to ensure it actually is a issue and if the suggestions results in better or worse code.

Reply:
I removed the last commit and énabled autocomplete .
It completed the PR before sonarqube could post an error status. :) 

But the found "issue" was a false positive since you cannot do "is" matching with typeof() since they are not compile time constants

Reply:
fyi: @SandstromErik you should be able to login to sonarcloud and resolve issues there (in this case as a false postive) and it will automatically edit the status of the PR.